src/server/main_service.cc

```diff
@@ -590,6 +600,42 @@ void Service::Init(util::AcceptServer* acceptor, std::vector<facade::Listener*>
   GenericFamily::Init(&pp_);
   server_family_.Init(acceptor, std::move(listeners), &cluster_family_);
 
+  if (string flag = GetFlag(FLAGS_replicaof); !flag.empty()) {
```

Comment:
I would extract all of this logic to a separate function and call it from here.

Reply:
Or better yet - use a custom-type flag and enforce the format there. See https://github.com/dragonflydb/dragonfly/commit/0fe6e36f255d798daee56dd70dec9a2602b64ae6 as an example

Reply:
I like this. Let's do it!